omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,108 @@
+
+@Resource(name = RestConstants.VERSION_1 + IdgenRestController.IDGEN_NAMESPACE + "/identifiersource", supportedClass = IdentifierSource.class, supportedOpenmrsVersions = {
+        "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*" , "2.1.*" })
+public class IdentifierSourceResource extends DelegatingCrudResource<IdentifierSource>{
+	
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		
+		DelegatingResourceDescription description = null;
+		
+		if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addRequiredProperty("uuid");
+			description.addRequiredProperty("name");
+			description.addRequiredProperty("description");
+			description.addRequiredProperty("identifierType");
+			
+			// links
+			description.addSelfLink();
+			if (rep instanceof DefaultRepresentation) {
+				description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			} 
+			
+		}
+		
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("name");
+		description.addRequiredProperty("description");
+		description.addRequiredProperty("identifierType");
+		
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoesNotSupportOperationException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("name");
+		description.addRequiredProperty("description");
+		description.addRequiredProperty("identifierType");
+		
+		return description;
+	}
+	
+	@Override
+	public IdentifierSource newDelegate() {
+		return new SequentialIdentifierGenerator();
+	}
+	
+	@Override
+	public IdentifierSource save(IdentifierSource delegate) {
+		return Context.getService(IdentifierSourceService.class).saveIdentifierSource(delegate);
+	}
+	
+	@Override
+	public IdentifierSource getByUniqueId(String uniqueId) {
+		return Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uniqueId);
+	}
+	
+	@Override
+    protected void delete(IdentifierSource delegate, String reason, RequestContext context) throws ResponseException {
+		Context.getService(IdentifierSourceService.class).retireIdentifierSource(delegate, reason);
+    }
+	
+	@Override
+	public void purge(IdentifierSource delegate, RequestContext context) throws ResponseException {
+		Context.getService(IdentifierSourceService.class).purgeIdentifierSource(delegate);
+	}
+	
+	@Override
+	public PageableResult doGetAll(RequestContext context)  throws ResponseException {
+		return new AlreadyPaged<IdentifierSource>(context, Context.getService(IdentifierSourceService.class).getAllIdentifierSources(false), false);
+	}
+	
```

Comment:
can you also add a display property getter?
@PropertyGetter("display")
	public String getDisplayString(IdentifierSource identifierSource) {
		return identifierSource.getIdentifierType() + " - " + identifierSource.getName();
	}

Reply:
added